tests/Http/PostTest.php

```diff
@@ -762,6 +762,70 @@ public function testPostsIndexAsNonAdminNonOwner()
     }
 
+    /**
+     * Test for retrieving all posts with some anonymous posts.
+     * Non-owners should not northstar_id or location.
+     *
+     * GET /api/v3/posts
+     * @return void
+     */
+    public function testPostsIndexWithAnonymousPosts()
+    {
+        // Create an accepted post.
+        $this->mockTime('8/3/2017 14:00:00');
+        $regularPost = factory(Post::class, 'accepted')->create();
+
+        // And then later, create 1 post that is anonymous.
+        $this->mockTime('8/3/2017 17:30:00');
+        $anonymousPost = factory(Post::class, 'accepted')->create();
+        $anonymousPost->actionModel->anonymous = 1;
+        $anonymousPost->actionModel->save();
+
+        // Anonymously hit the endpoint and northstar_id should not be returned for the anonymous post (first post since it was most recently created).
+        $response = $this->getJson('api/v3/posts');
+        $response->assertStatus(200);
+
+        $this->assertArrayNotHasKey('northstar_id', $response->decodeResponseJson()['data'][0]);
+        $this->assertEquals($regularPost->northstar_id, $response->decodeResponseJson()['data'][1]['northstar_id']);
+
+        // Hit the endpoint with access credentials from another user and should have same results as above.
+        $response = $this->withAccessToken($regularPost->northstar_id)->getJson('api/v3/posts');
+        $response->assertStatus(200);
+        $this->assertArrayNotHasKey('northstar_id', $response->decodeResponseJson()['data'][0]);
+        $this->assertEquals($regularPost->northstar_id, $response->decodeResponseJson()['data'][1]['northstar_id']);
+
+        // Hit the endpoint with filter[northstar_id] and should have same results as above.
+        $response = $this->getJson('api/v3/posts?filter[northstar_id]=' . $anonymousPost->northstar_id);
+        $response->assertStatus(200);
+        $this->assertArrayNotHasKey('northstar_id', $response->decodeResponseJson()['data']);
+
+        $response = $this->withAccessToken($regularPost->northstar_id)->getJson('api/v3/posts?filter[northstar_id]=' . $anonymousPost->northstar_id);
+        $response->assertStatus(200);
+        $this->assertArrayNotHasKey('northstar_id', $response->decodeResponseJson()['data']);
+
+        // Hit the endpoint as the owner of the post and you should be able to see northstar_id for anonymous post.
+        $response = $this->withAccessToken($anonymousPost->northstar_id)->getJson('api/v3/posts');
+        $response->assertStatus(200);
+        $this->assertEquals($anonymousPost->northstar_id, $response->decodeResponseJson()['data'][0]['northstar_id']);
+        $this->assertEquals($regularPost->northstar_id, $response->decodeResponseJson()['data'][1]['northstar_id']);
+
+        // Hit the endpoint with filter[northstar_id] and should have same results as above.
+        $response = $this->withAccessToken($anonymousPost->northstar_id)->getJson('api/v3/posts?filter[northstar_id]=' . $anonymousPost->northstar_id);
```

Comment:
This one's a bit tricker – the `northstar_id` on the post is technically exposed by the fact that it showed up on a `?filter[northstar_id]=…` response (since anything returned by this query would _have_ to have that user ID). Ideally, we'll want to update the index route to specifically exclude "anonymous" posts if querying for a user that's not you.

Reply:
ah great catch! 

Updated the tests and I tried to add it to the `whereVisible` query scope but it is still being returned when it shouldn't be. Using clockwork, it looks like the query is correct yet I'm not sure why it is still returned. Am I missing something? 

![screen shot 2019-02-21 at 3 34 37 pm](https://user-images.githubusercontent.com/9019452/53199927-4d7c6f80-35ee-11e9-9297-92f4f4b8f298.png)

Reply:
Moved this to a separate query scope and got it to work! Ready for review!

Reply:
oh wait - this is omitting the entire thing from the index without this filter - need to update this! I'll let you know when it is ready for review :)

Reply:
ok NOW this is ready! :)